File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mx=0,m=Integer.MAX_VALUE;
        for(int i=0;i<prices.length;i++)
        {
            if(m>prices[i])
            m=prices[i];
            if(prices[i]-m>mx)
            mx=prices[i]-m;
            
        }
        return mx;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n=prices.size(),mx=prices[n-1];
        vector<int> temp(n);
        for(int i=n-1;i>=0;i--)
        {
            mx=max(mx,prices[i]);
            prices[i]=mx-prices[i];
        }
        return *max_element(prices.begin(),prices.end());
    }
};